build-test.cmd

```diff
@@ -6,8 +6,17 @@ set __BuildArch=x64
 set __VCBuildArch=x86_amd64
 set __BuildType=Debug
 set __BuildOS=Windows_NT
-set __VSVersion=vs2015
-set __VSToolsRoot=%VS140COMNTOOLS%
+
+:: Default to highest Visual Studio version available
+if defined VS150COMNTOOLS (
```

Comment:
This is how it is today. VS150COMNTOOLS is only defined if either:
A) The user has manually defined it or
B) The user is running from a VS2017 Developer Command Prompt

In either case, I would argue the user has explicitly opted into building using the VS2017 toolset.